### PR TITLE
Feature/add weight in chapter children

### LINKED ISSUE #359: Add "weight" in chapter children
## Motivations
Some item children in a chapter may have different weight when computing the score of a chapter.
These score weight have to be editable. (cfr [design](https://france-ioi.github.io/algorea-designs/03.Activities_Edit_05.Edit-chapter-content) and [design branch](http://dev.algorea.org/branch/design))

## Subtasks
In item edit, "Content" tab:
* On the top right of the children list (only if loaded, not empty, not in error), display a switch "Enable score weight".  Its default value is whether there is at least one of the children with a `score_weight` != 1. 
* If enabled, display in each row the current weight in the form `<weight>x` (e.g., "2x", "10x"). If clicked, this label becomes an editable input accepting only digits (between 0 and 999, both included).  --> same behavior as in the [design branch](http://dev.algorea.org/branch/design) (you can copy code from this branch as well)
* On save, update the weights. If "Enable score weight" is disable, set "1" to all `score_weight`'s

### PR DESCRIPTION
**Pull request by Iloveall**
Fixes #359

### CODE REVIEW

**Review by smadbe — CHANGES_REQUESTED** [review]
In this chapter, I have put one weight at 0. However if I refresh, it does not enable weights.

http://dev.algorea.org/branch/feature/add-weight-in-chapter-children/en/#/activities/by-id/1012286565380301759;attempId=0;path=4702,4102/edit

**Review by smadbe — CHANGES_REQUESTED** [review]
If I add a new child, it does not display any weight:
Screenshot 2021-04-29 at 10 42 13

File: src/app/modules/shared-components/components/add-content/add-content.component.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export interface AddedContent<T> {
   id?: string,
   title: string,
   type: T,
+  scoreWeight?: number,
```
Comment (smadbe) [inline src/app/modules/shared-components/components/add-content/add-content.component.ts:11]: This component has nothing to do with the weight. It is also used for groups where there is no weight. Weight should just be added with the data is received, so in `addChild` of item-children-edit.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/item/components/item-children-edit/item-children-edit.component.ts [hunk 2]
```diff
@@ -30,6 +31,7 @@ export class ItemChildrenEditComponent implements OnChanges {
   state: 'loading' | 'error' | 'ready' = 'ready';
   data: ChildData[] = [];
   selectedRows: ChildData[] = [];
+  scoreWeightEnable = false;
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/components/item-children-edit/item-children-edit.component.ts]: Fixed

File: src/app/modules/item/http-services/create-item.service.ts [hunk 3]
```diff
@@ -13,6 +13,7 @@ interface NewItemData {
 export type NewItem = {
   title: string,
   type: ItemType,
+  scoreWeight?: number,
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/http-services/create-item.service.ts]: Fixed

File: src/app/modules/shared-components/components/add-content/add-content.component.ts (hunk 1 above)
Comment (Iloveall) [reply src/app/modules/shared-components/components/add-content/add-content.component.ts]: Fixed

File: src/app/modules/item/components/item-children-edit/item-children-edit.component.ts [hunk 4]
```diff
@@ -4,11 +4,14 @@ import { GetItemChildrenService } from '../../http-services/get-item-children.se
 import { Subscription } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { ItemType } from '../../../../shared/helpers/item-type';
+import { DEFAULT_SCORE_WEIGHT } from '../../http-services/create-item.service';
+import { AddedContent } from '../../../shared-components/components/add-content/add-content.component';
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/components/item-children-edit/item-children-edit.component.ts:7]: Fixed

**Pull request merged**